### DISCUSSION TITLE
[FEATURE REQUEST] Alignment PCB-Exposure

### DISCUSSION THREAD

**Discussion by rdu** [sn4k3/UVtools#494] [Ideas]
## Is your feature request related to a problem? Please describe

I use UVtools to expose my PCBs. This is a unique feature that I'm really incredibly happy about because it speeds up my workflow significantly.

However, I have a problem...

When I try to expose the soldermask, the copper traces are not in line with the holes in the soldermask. If both were in the same spot, then I could just align the PCB twice to the corner of the screen and would have a perfect overlay.

Now I have to make it such that I export the two layers each as, manually overlay them with Gimp, and then import them back again.

My guess is that there is no board size specified in the Gerber format, but maybe the board outline layer could be used as a reference?

Perfect would be to be able to select the start coordinates and a repetition with offset (panalize). (I know, this is actually already a separate features request).

**Comment by sn4k3** [comment]
The start x,y is always 0, the left,top padding come from pcb placement that should be respected. However i never test for soldermask layer. Zip one sample and send me so i can track down the problem.

Your workflow is expose copper and soldermask while not moving the board?

**Comment by rdu** [comment]
After etching, I expose the soldermask ink by moving the board to the exact same spot. I printed myself a template so that I only have to push the board into the top right corner.

I have found my mistake. The program works perfectly. I overlooked that the 2nd time the mirror feature (which I had previously turned off) was on again. I then misinterpreted the resulting image as a misplaced.

Nevertheless, it would be nice if one could repeat the picture. And maybe the mirror button could remember its position. I think I will forget this quite often.... ;-)

**Comment by sn4k3** [comment]
> it would be nice if one could repeat the picture.

Tools - Pattern

> And maybe the mirror button could remember its position

Save a profile just with mirror off and set it as default

**Comment by rdu** [comment]
Thank you!

**Reply by sn4k3** [reply]
Do you think importing multiple gerber and merge them into one layer have any possible use?

**Reply by rdu** [reply]
I can think of a use case (at least for drilling) - Sometimes it is difficult for me to see where to drill when drilling manually. I then print myself a template with a laser printer on foil and then drill with it. However, it's not always perfect because chips get between the board and the foil. Maybe you could leave out some copper at the drilling points (with adjustable percentage size) then you can see the holes directly. You could simply create a solid circle at the drill points and then calculate the difference from the original layer.

It would also be nice if you could open archives with Gerber files and select the layer. I use easy-eda and always have a ZIP archive that I have to unpack first (and then rename the files to *.gbr).
But of course this is not a problem - it would just be "nice to have".

**Reply by rdu** [reply]
Hello,

I have noticed that you have done a lot of work ;-)  Thanks for that. I think it's great that you can now open ZIP archives. However, the files in it must be named *.gbr, otherwise they are not displayed. 
I use easyeda - and there the files have different extensions. I have attached a small example for you. It would be really cool if I didn't have to rename it all the time
[Gerber_PCB_Weather-top_2022-06-24 (1).zip](https://github.com/sn4k3/UVtools/files/8978980/Gerber_PCB_Weather-top_2022-06-24.1.zip)
.

**Reply by sn4k3** [reply]
I can add those extensions, the DRL however can't be added as it's another format

**Reply by sn4k3** [reply]
(Hint) With UVtools you can create a PCB trace image like x-ray just for fun or keep record: 
![poussin zcode heatmap](https://user-images.githubusercontent.com/113281/175659614-5c2f5fea-6002-4367-80ce-1aae65e3ad82.png)

A more complex sample:

![poussin zcode heatmap](https://user-images.githubusercontent.com/113281/175660072-a375a66e-9da0-4987-934b-256a0b112ac7.png)
